File: app/src/main/java/lt/vilnius/tvarkau/FullscreenImageActivity.java
package lt.vilnius.tvarkau;

import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.RelativeLayout;

import butterknife.BindView;
import butterknife.ButterKnife;
import lt.vilnius.tvarkau.views.adapters.FullscreenImagesPagerAdapter;
import lt.vilnius.tvarkau.views.adapters.HackyViewPager;

public class FullscreenImageActivity extends BaseActivity {

    public final static String EXTRA_IMAGE_POSITION = "FullscreenImageActivity.position";
    public final static String EXTRA_PHOTOS = "FullscreenImageActivity.photos";

    @BindView(R.id.problem_images_view_pager)
    HackyViewPager problemImagesViewPager;
    @BindView(R.id.fullscreen_layout)
    RelativeLayout fullscreenLayout;
    @BindView(R.id.toolbar)
    Toolbar toolbar;

    private int initialImagePosition;
    private String[] photos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        View decorView = getWindow().getDecorView();
        decorView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN |
                View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);

        setContentView(R.layout.activity_fullscreen);

        ButterKnife.bind(this);

        setSupportActionBar(toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setTitle("");
            getSupportActionBar().hide();
        }

        if (getIntent().getExtras() != null) {
            initialImagePosition = getIntent().getExtras().getInt(EXTRA_IMAGE_POSITION);
            photos = getIntent().getExtras().getStringArray(EXTRA_PHOTOS);
        }

        initializePager();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void initializePager() {
        problemImagesViewPager.setAdapter(new FullscreenImagesPagerAdapter<>(this, photos, new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (FullscreenImageActivity.this.getSupportActionBar() != null) {
                    if (FullscreenImageActivity.this.getSupportActionBar().isShowing()) {
                        FullscreenImageActivity.this.getSupportActionBar().hide();
                    } else {
                        FullscreenImageActivity.this.getSupportActionBar().show();
                    }
                }
            }
        }));
        problemImagesViewPager.setOffscreenPageLimit(3);
        problemImagesViewPager.setCurrentItem(initialImagePosition);
    }
}

File: app/src/main/java/lt/vilnius/tvarkau/FullscreenImageActivity.kt
package lt.vilnius.tvarkau

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import kotlinx.android.synthetic.main.activity_fullscreen.*
import lt.vilnius.tvarkau.views.adapters.FullscreenImagesPagerAdapter

class FullscreenImageActivity : BaseActivity() {

    private val initialImagePosition: Int
        get() = intent.getIntExtra(EXTRA_IMAGE_POSITION, 0)

    private val photos: Array<String>
        get() = intent.getStringArrayExtra(EXTRA_PHOTOS) ?: emptyArray()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        window.decorView.systemUiVisibility =
                View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN

        setContentView(R.layout.activity_fullscreen)

        setSupportActionBar(toolbar)
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            title = ""
            hide()
        }

        initializePager()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    private fun initializePager() {
        problem_images_view_pager.adapter = FullscreenImagesPagerAdapter(this, photos) {
            supportActionBar?.apply {
                if (isShowing) {
                    hide()
                } else {
                    show()
                }
            }
        }
        problem_images_view_pager.offscreenPageLimit = 3
        problem_images_view_pager.currentItem = initialImagePosition
    }

    companion object {
        val EXTRA_IMAGE_POSITION = "FullscreenImageActivity.position"
        val EXTRA_PHOTOS = "FullscreenImageActivity.photos"
    }
}